Ensure composer patches fail the build if a patch fails

cweagans/composer-patches has a setting to ensure the build fails if any patch fails: `COMPOSER_EXIT_ON_PATCH_FAILURE=1`.

We would like this to be handled via a composer.json config value. We do not know if the composer-patches library can receive it that way, though it seems that proper use of the composer config API would allow it to be injected either way.

### Action Items

1. Check to see if composer.json config can be used to activate this setting.
2. If not PR upstream and see if it will have a release quickly enough to leverage.
3. If a release with a fix is fast enough, add that setting to our composer.json config.
4. If a release is not fast enough, add the environment variable to our build.yml and build.devcloud.yml templates.

This ticket emerged from conversation with @jhedstrom and @scottalan

I placed the variable in the wrong container. It should live in the `base` service. I have tested this and adding `COMPOSER_EXIT_ON_PATCH_FAILURE: 1` does in fact work.

```
Running "composer:install" (composer) task
Gathering patches for root package.
Removing package drupal/core so that it can be re-installed and re-patched.
Deleting build/html/core - deleted
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Package operations: 1 install, 1 update, 0 removals
Gathering patches for root package.
Gathering patches for dependencies. This might take a minute.
  - Installing drupal/core (8.3.7): Loading from cache
  - Applying patches for drupal/core
    https://www.drupal.org/files/issues/2699157-23.drupal.Plugin-Lazy-loading-can-cause-usort-warning.patch (2699157 - Plugin Lazy loading can cause usort warning)
    https://www.drupal.org/files/issues/2759397-1-entity_reference_recursion.patch (2759397 - Patch EntityReferenceItemNormalizer to prevent recursion)
    https://www.drupal.org/files/issues/2679775-11-inline-labels.patch (2679775 - Fixes float issue with inline label fields (entity references on most cases).)
    https://www.drupal.org/files/issues/duplicate_ajax_wrapper-2346893-194.patch (2346893 - Duplicate AJAX wrapper around a file field)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/duplicate_ajax_wrapper-2346893-194.patch


  [Exception]
  Cannot apply patch 2346893 - Duplicate AJAX wrapper around a file field (https://www.drupal.org/files/issues/duplicate_ajax_wrapper-2346893-194.patch)!


install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--] [<packages>]...

Warning: Task "composer:install" failed. Use --force to continue.

Aborted due to warnings.
[ERROR] Error running project script 'install': 3
```

--- a/generators/environment/templates/docker/build.yml
+++ b/generators/environment/templates/docker/build.yml
@@ -169,6 +169,7 @@ services:
       SIMPLETEST_DB: mysql://admin:admin@db/<%= machineName %>_drupal
       SIMPLETEST_URL: <%= host.local %>
       BROWSERTEST_OUTPUT_DIRECTORY: /tmp
+      COMPOSER_EXIT_ON_PATCH_FAILURE: 1
 
 volumes:
   # This defines the filesystem synchronization volume used for application

